.github/workflows/build-image.yml

```diff
@@ -0,0 +1,41 @@
+name: Build Vigilo image and push to ECR
+on:
+   push:
+      branches:
+        - main
+    # Allow running this workflow manually from the Actions tab
+    workflow_dispatch:
+
+jobs:
+   build_image:
+      name: Build Vigilo image and push to ECR
+      runs-on: ubuntu-latest
+      permissions:
+        id-token: write
+        contents: read
+
+      steps:
+       - name: Checkout
+         uses: actions/checkout@v3
+
+       - name: Configure AWS credentials
+         uses: aws-actions/configure-aws-credentials@v2
+         with:
+            role-to-assume: ${{ env.DEPLOY_IAM_ROLE }}
+            aws-region: us-east-1
+
+       - name: Login to Amazon ECR
+         id: login-ecr
+         uses: aws-actions/amazon-ecr-login@v1
+
+       - name: Build, Tag, and Push Image to Amazon ECR
+         env:
+           REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+           REPOSITORY: vigilo
+           DOCKER_TAG: ${{ github.sha }}
+         run: |
+           docker build -t $REGISTRY/$REPOSITORY:$DOCKER_TAG -f Dockerfile .
+           docker tag $REGISTRY/$REPOSITORY:$DOCKER_TAG $REGISTRY/$REPOSITORY:latest
```

Comment:
Hmm, for added security I am wondering if we should log the deployment process in Slack to keep track of when these deployments will occur since the workflow could be ran manually.

Reply:
> Hmm, for added security I am wondering if we should log the deployment process in Slack to keep track of when these deployments will occur since the workflow could be ran manually.

I am not sure if this is necessary. Maybe we can tighten up the repo settings? Also we can remove the `workflow_dispatch` if needed.

Reply:
Then let's just disallow the `workflow_dispatch` trigger for now

Reply:
Updated in https://github.com/iFixit/vigilo/pull/18/commits/acf505f76ae0affc5f4d7fbbd8d30273cdd68ee0.